### PR TITLE
Add support for passing a custom class name to all components

### PR DESCRIPTION
**Pull request by HenriBeck**
This adds support for passing a `className` prop to all of the exposed components.

Fixes: #210

### CODE REVIEW

**Comment by nolanlawson** [comment]
Just for context, what problem does this solve? If you'd like to style the emoji-picker, you can always use a class selector that relies on a parent (or some other technique). For instance:

```html
<div class="parent">
  <div class="emoji-mart">
    <!-- etc -->
  </div>
</div>
```

```css
.parent .emoji-mart {
  /* etc */
}
```

 Is there a case where you need to set a class on the emoji-picker itself?

**Comment by HenriBeck** [comment]
Just wrapping the picker around another DOM node for styling the picker is in my opinion bad practice.

Also, this could potentially break some layouts because of the added node.

File: src/utils/shared-props.js [hunk 1]
```diff
@@ -37,9 +38,6 @@ const EmojiDefaultProps = {
   tooltip: false,
   backgroundImageFn: (set, sheetSize) =>
     `https://unpkg.com/emoji-datasource-${set}@${EMOJI_DATASOURCE_VERSION}/img/${set}/sheets-256/${sheetSize}.png`,
-  onOver: () => {},
-  onLeave: () => {},
-  onClick: () => {},
```
Comment (nolanlawson) [inline src/utils/shared-props.js]: Why remove these onOver/onLeave/onClicks?

File: src/utils/shared-props.js [hunk 2]
```diff
@@ -105,7 +102,6 @@ const PickerDefaultProps = {
   autoFocus: false,
   custom: [],
   skinEmoji: '',
-  notFound: () => {},
```
Comment (nolanlawson) [inline src/utils/shared-props.js]: Ditto

**Comment by EtienneLem** [comment]
> Just wrapping the picker around another DOM node for styling the picker is in my opinion bad practice.

I’m curious why that would be. It’s arguably exactly what cascading means 😄

> Also, this could potentially break some layouts because of the added node.

This will not break emoji-mart layout. As for the layout around emoji-mart, I’m not sure I see how adding emoji-mart doesn’t break the layout, but having emoji-mart inside let’s say another `<div>` would 🤔

If I’m being honest, I feel like this is a little bit superfluous. (It also doesn’t add new className to all the existing `emoji-mart-*` classes)

**Comment by nolanlawson** [comment]
On second thought, yeah I agree with @EtienneLem. It's not like you have to add a second `<div>` - there must be _some_ element wrapping `emoji-mart`, which you can use to style it. If it's the only `emoji-mart` on your page, then it's even easier - just add your own CSS to target the `.emoji-mart` classes.

Plus adding this to the codebase means one more thing to maintain and document... I would argue for resolving this as wontfix. Thanks very much for the contribution though; it highlights the fact that perhaps we could do a better job documenting how to style emoji-mart.

**Comment by topolanekmartin** [comment]
> 
> 
> > Just wrapping the picker around another DOM node for styling the picker is in my opinion bad practice.
> 
> I’m curious why that would be. It’s arguably exactly what cascading means 😄
> 
> > Also, this could potentially break some layouts because of the added node.
> 
> This will not break emoji-mart layout. As for the layout around emoji-mart, I’m not sure I see how adding emoji-mart doesn’t break the layout, but having emoji-mart inside let’s say another `<div>` would 🤔
> 
> If I’m being honest, I feel like this is a little bit superfluous. (It also doesn’t add new className to all the existing `emoji-mart-*` classes)

I am sorry but I don't agree with your opinion mess up DOM with another node only for styling your component. It is not good for many reasons. What is so wrong with adding property className to the component's root element?

**Comment by EtienneLem** [comment]
> […] mess up DOM with another node only for styling your component.

You don’t have to. You can target `.emoji-mart` from your own stylesheets to customize 100% of EmojiMart styles.

**Pull request closed without merging**